bundles/azure/aso/psql/bundle/config/01-flexible-server.ytt.yml

```diff
@@ -0,0 +1,229 @@
+    kapp.k14s.io/noop: ""
+spec:
+  azureName: #@ data.values.resource_group
+  location: #@ data.values.location
+
+#@ databaseCounter = 0
+#@ for db in data.values.databases:
+#@   dbName = db.name
+#@   if dbName == "":
+#@     dbName = flexibleServerName + "-" + str(databaseCounter)
+#@     databaseCounter += 1
+#@   end
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersDatabase
+metadata:
+  name: #@ dbName
+  namespace: #@ data.values.resource_namespace
+spec:
+  azureName: #@ dbName
+  owner:
+    name: #@ flexibleServerName
+  charset: #@ db.charset
+  collation: #@ db.collation
+#@ end
+
+#@ fwRuleCounter = 0
+#@ for val in data.values.firewall_rules:
+#@   startAddr = ip.parse_addr(val.startIp)
+#@   endAddr = ip.parse_addr(val.endIp)
+--- 
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServersFirewallRule
+metadata:
+  name: #@ data.values.resource_name + "-" + str(fwRuleCounter)
+  namespace: #@ data.values.resource_namespace
+spec:
+  owner:
+    name: #@ flexibleServerName
+  startIpAddress: #@ startAddr.string()
+  endIpAddress: #@ endAddr.string()
+#@   fwRuleCounter += 1
+#@ end  
+---
+apiVersion: dbforpostgresql.azure.com/v1beta20210601
+kind: FlexibleServer
+metadata:
+  name: #@ flexibleServerName
+  namespace: #@ data.values.resource_namespace
+spec:
+  location: #@ data.values.location
+  azureName: #@ flexibleServerName
+  owner:
+    name: #@ resourceGroupK8SName
+  version: #@ data.values.version
+  sku:
+    name: #@ data.values.instance_type
+    tier: #@ data.values.instance_tier
+  administratorLogin: #@ data.values.administrator_name
+  administratorLoginPassword:
+    name: #@ dbSecretName
+    key: password
+  storage:
+    storageSizeGB: #@ data.values.instance_storage_size_gb
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.resource_namespace
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+- apiGroups:
+  - dbforpostgresql.azure.com
+  resources:
+  - flexibleservers
+  - flexibleserversdatabases
+  - flexibleserversfirewallrule
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+- apiGroups:
+  - resources.azure.com
+  resources:
+  - resourcegroups
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.resource_name
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: #@ roleName
+  namespace: #@ data.values.resource_namespace
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: #@ roleName
+subjects:
+- kind: ServiceAccount
+  name: #@ serviceAccountName
+  namespace: #@ serviceAccountNamespace
+---
+apiVersion: secretgen.carvel.dev/v1alpha1
```

Comment:
So when we allow to create multiple databases, I think we want to have multiple claimable secrets. There was some discussion about that earlier (https://github.com/vmware-tanzu/tanzu-application-platform-reference-service-packages/pull/11#discussion_r921495607) and the question what we do about that (and further, what to do when we also allow multiple users) has not been answered yet. Thus we, for now, decided to not make the databases configurable yet, but just create one default database.
I think we need to figure out what we want to do about support of multiple databases / users soon.

Reply:
To be honest, we added multiple database support this morning, and I think we forgot that the secret contains the database name. 

The credentials are on the FlexibleServer, so that part is safe. Although I think the select code snippet is of?

Reply:
> Although I think the select code snippet is of?

Not entirely sure, what you mean by that. If it's about getting the `db` input resources, I think this will only work if users explicitly create a database named the same as the `${resource_name}`; otherwise the package will create a database named `${resource_name}-1` (if I read the code correctly?).
And if that's the case, the SA in the `SecretTemplate` won't be able to get/read the `FlexibleServersDatabase/${resource_name}`.

My [other comment](https://github.com/vmware-tanzu/tanzu-application-platform-reference-service-packages/pull/22/files#r969798929) is exactly about that too.

----

However, my comment was more about: "Should we support multiple databases (and if so: how?)", similar to Sam's comment in https://github.com/vmware-tanzu/tanzu-application-platform-reference-service-packages/pull/22/files#r969826418.

If we want to support multiple databases, we sure need a `SecretTemplate` per database. But I think it is still a point of discussion if we want to support that (for the initial version of the package(s)).

Reply:
> If we want to support multiple databases, we sure need a SecretTemplate per database. But I think it is still a point of discussion if we want to support that (for the initial version of the package(s)).

Point taken, and agree that for a first version this can be postponed (and the feature rolled backed for now).

I do believe the feedback we've gotten (via Timo and Srikanth) requested the ability to create more than one database per server. So it is something to tackle sooner rather than later.

Regarding the code snippet, the snippet this comment is one seems to have nothing to do with your comment?
```
  name: #@ serviceAccountName
  namespace: #@ serviceAccountNamespace
---
apiVersion: secretgen.carvel.dev/v1alpha1
```

Reply:
> [...] ability to create more than one database per server.

We'll definitely have to implement this at some point. Another thing we've discussed is: How to support multiple users on the same database. It's not hard to imagine use-cases where multiple different users (potentially with different permissions) would want to access the same database.

> Regarding the code snippet, the snippet this comment is one seems to have nothing to do with your comment?

I meant to flag, that we should create multiple `SecretTemplate`s, if we want to support multiple databases. My bad, the line below might have been a better place.

Reply:
>I do believe the feedback we've gotten (via Timo and Srikanth) requested the ability to create more than one database per server. So it is something to tackle sooner rather than later.

One thing we've considered is having multiple packages for this usecase. E.g. A "database" package that refers to a "server" package. That way you can manage the lifecycle and get credentials of a database separately. We'd have to think more about how this works though.

Reply:
To be clear, these are all good points. We have decided to remove the multiple database support for now, and address this in the future with a more thought through solution.